ocaml/typing/typecore.ml

```diff
@@ -4614,13 +4942,8 @@ and type_expect_
       let (pat_exp_list, new_env) =
         type_let existential_context env rec_flag spat_sexp_list allow_modules
       in
-      let in_function =
-        match sexp.pexp_attributes with
-        | [{Parsetree.attr_name = {txt="#default"};_}] -> in_function
```

Comment:
It was added in https://github.com/ocaml-flambda/ocaml-jst/commit/66d811078463cbdcb55b3d6d35849fc80b3ba995#diff-154f81eaa30645133fc97f3cd10102b14c29295eff4e36cb8982a33ae884b3d7, I believe because modes warrant more precise tracking of `in_function`. So this seems fine to me, but thanks for checking.